### ISSUE TITLE
Releasing the same version twice makes the action stick onto that version

### ISSUE THREAD

**Issue by Ghustavh97** [google-github-actions/release-please-action#409]
### TL;DR

If you `git commit --allow-empty -m "chore: release 2.0.0" -m "Release-As: 2.0.0"` by mistake and version 2.0.0 already exists you will get this message

```
✔ creating release v2.0.0
Error: release-please failed: Validation Failed: {"resource":"Release","code":"already_exists","field":"tag_name"}
```

Which is fine, but if you try to release the correct version by e.g `git commit --allow-empty -m "chore: release 3.0.0" -m "Release-As: 3.0.0"`  release please will try to release version `v2.0.0` again.

### Expected behavior

Release please should not keep trying to release an already existing version when the commit message contains a different version.

### Observed behavior

WIll keep trying to release the same version over and over again, see test repo https://github.com/Ghustavh97/release-please-test

### Action YAML

```yaml
name: CD/Release

on:
  push:
    branches:
      - master
jobs:
  release-please:
    runs-on: ubuntu-latest
    steps:
      - uses: GoogleCloudPlatform/release-please-action@v3
        id: release
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          release-type: node
          package-name: g97-release-please-test
      # The logic below handles the npm publication:
      - uses: actions/checkout@v2
        # these if statements ensure that a publication only occurs when
        # a new release is created:
        if: ${{ steps.release.outputs.release_created }}
      - uses: actions/setup-node@v1
        with:
          node-version: 12
          registry-url: 'https://registry.npmjs.org'
        if: ${{ steps.release.outputs.release_created }}
      # if you are using Yarn, substitute the command below with `yarn install --frozen-lockfile`
      - run: npm ci
        if: ${{ steps.release.outputs.release_created }}
      - run: npm publish
        env:
          NODE_AUTH_TOKEN: ${{secrets.NPM_AUTH_TOKEN}}
        if: ${{ steps.release.outputs.release_created }}
```

**Comment by jcornaz** [comment]
I ended up with the same problem. Alghough not really for the same reason. I think in my case, release-please just messed up because I used pre-releases. 

Anyway release-please is now stuck an always fail: https://github.com/jcornaz/benimator/runs/4927482401?check_suite_focus=true

**Comment by SurferJeffAtGoogle** [comment]
This issue will fall out of SLO soon.
